src/index.js

```diff
@@ -104,6 +128,12 @@ function ParseServer({
     DatabaseAdapter.setAppDatabaseURI(appId, databaseURI);
   }
   
+  if (verifyUserEmails && !publicServerURL && !process.env.TESTING) {
+    console.warn("");
+    console.warn("You should set publicServerURL to serve the public pages");
```

Comment:
I think this should probably be an error that goes into validateEmailConfiguration function. Verification emails can't possibly work if you don't have a public server URL, can they? We could mention that for development, it's OK to use a fake or localhost URL.